### PR TITLE
Homepage tests

### PR DESCRIPTION
**Pull request by martypowell**
Big changes and no so much.

Methods use named functions.

Everything else is es6 format, the es6 class made things a bit clunky and hard to follow, and needs more work. (it may not even need to be a class). Would also like to create a base class for automaters eventually.

### CODE REVIEW

File: debug.log [hunk 1]
```diff
@@ -0,0 +1,30 @@
+[1113/154121.697:ERROR:process_reader_win.cc(123)] NtOpenThread: {Access Denied} A process has requested access to an object, but has not been granted those access rights.  (0xc0000022)
```
Comment (mxsnyder) [inline debug.log:1]: .gitignore maybe?

File: spec/features/homepage/homepage-carousel.automater.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	getActiveImgSrc(automater) {
+		return new Promise((resolve, reject) =>
+			automater.driver.findElement(By.css(automater.carouselActiveItemImageSelector))
+				.then(elm => elm.getAttribute('src')
+					.then(resolve)
+					.catch(reject))
+			.catch(reject));
+	}
+
+	static getCssValueByElement(elm, prop) {
```
Comment (mxsnyder) [inline spec/features/homepage/homepage-carousel.automater.js]: To keep things organized, would you keep your static methods together if possible?

File: spec/features/homepage/homepage-carousel.spec.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	it('Should pause rotation if the mouse is hovered over an item', (done) => {
+		automater.isCarouselPaused(10000)
+			.then((isCarouselPaused) => {
+				expect(isCarouselPaused).to.equal(true);
+				done();
+			})
+			.catch(err => handleException(err, done));
+	});
+
+	before(() => {
```
Comment (mxsnyder) [inline spec/features/homepage/homepage-carousel.spec.js:54]: I think it makes more sense to have before(), beforeEach() and after() before your it()'s, since that's valuable info you would want to know about as you're scrolling through the tests.

**Review by mxsnyder — APPROVED** [review]
This looks great, my only comments are organizational, and aren't critical. Nice work using class!

**Comment by martypowell** [comment]
did not push my changes.
